VimeoUpload/Upload/Descriptor System/Cameo/CAMUploadDescriptor.swift

```diff
@@ -0,0 +1,356 @@
+//
+//  CAMUploadDescriptor.swift
+//  Cameo
+//
+//  Created by Westendorf, Michael on 6/27/16.
+//  Copyright © 2016 Vimeo. All rights reserved.
+//
+
+import Foundation
+import AFNetworking
+import VimeoNetworking
+
+public class CAMUploadDescriptor: ProgressDescriptor, VideoDescriptor
+{
+    let videoUrl: NSURL
+    let videoSettings: VideoSettings?
+    let thumbnailFileUrl: NSURL?
+    
+    private(set) var uploadTicket: VIMUploadTicket?
+    private(set) var video: VIMVideo?
+    private(set) var thumbnailTicket: VIMThumbnailUploadTicket?
+    private(set) var thumbnail: VIMPicture?
+    
+    private (set) var currentRequest = CAMUploadRequest.CreateVideo
+    {
+        didSet
+        {
+            print("\(self.currentRequest.rawValue) \(self.identifier)")
+        }
+    }
+ 
+    private struct ArchiverConstants {
+        static let VideoUrlKey = "videoUrl"
+        static let ThumbnailFileUrlKey = "thumbnailFileUrl"
+        static let VideoSettingsKey = "videoSettings"
+        static let UploadTicketKey = "uploadTicket"
+        static let CurrentRequestKey = "currentRequest"
+    }
+    
+    //MARK: VideoDescriptor Protocol
+    
+    public var type: VideoDescriptorType
+    {
+        return .Upload
+    }
+    
+    public var videoUri: VideoUri?
+    
+    public var progressDescriptor: ProgressDescriptor
+    {
+        return self
+    }
+    
+    //MARK: Initializers
+    
+    required public init()
+    {
+        fatalError("default init() should not be used, use init(videoUrl:videoSettings:thumbnailUrl:) instead")
+    }
+    
+    public init(videoUrl: NSURL, videoSettings: VideoSettings? = nil, thumbnailFileUrl: NSURL?)
+    {
+        self.videoUrl = videoUrl
+        self.videoSettings = videoSettings
+        self.thumbnailFileUrl = thumbnailFileUrl
+        
+        super.init()
+    }
+    
+    //MARK: Overrides
+    
+    override func prepare(sessionManager sessionManager: AFURLSessionManager) throws
+    {
+        let sessionManager = sessionManager as! VimeoSessionManager
+        
+        do
+        {
+            try self.transitionToState(request: .CreateVideo, sessionManager: sessionManager)
```

Comment:
Do we need to transition to state .CreateVideo if that's what the `self.currentRequest` var is initialized to?